master/buildbot/db/changes.py

```diff
@@ -195,7 +195,8 @@ def getChangesForBuild(self, buildid):
 
         # For each codebase, append changes until we match the parent
         for cb, change in iteritems(fromChanges):
-            if change and change['changeid'] != toChanges.get(cb, {}).get('changeid'):
+            toCbChange = toChanges.get(cb) or {}  # Careful; toChanges[cb] may be Nnoe from getChangeFromSSid
```

Comment:
There is a typo in the comment + the line is > 110 char. 
Please put the comment on the previous line

looks good apart from that.